feature/issue-39: Expose shapefile subsetting to Harmony

skorper opened the pull request:
Github Issue: #39

### Description

Expose shapefile subsetting to Harmony

### Overview of work done

- Updated `subset_harmony.py` to check for `shape` param and pass in if present.
- In the subsetter, check for shapefile before bbox. This is necessary because the Harmony wrapper will always pass in a bbox (but maybe global).
  - If both bbox and shapefile are provided, the bbox will be used by Harmony to filter granules but the shapefile is used in l2ss-py for the subsetting itself.
 
### Overview of verification done

Still putting together a unit test

### Overview of integration done

Tested with local Harmony deployment with collection `ASCATB-L2-Coastal`.

## PR checklist:

* [X] Linted
* [ ] Updated unit tests
* [X] Updated changelog
* [X] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

skorper commented:
> The transformation train demo revealed there may be an easier way to do this, where the service always receives GeoJSON. I will get more info and update this PR as needed.

I've looked into this. It seems that Harmony will convert the shapefile to GeoJSON automatically, which I was not aware of and did not reflect accurately during the demo. However our code already supports GeoJSON (to my surprise) so no change is needed by us.

frankinspace commented on README.md:
```diff
@@ -10,6 +10,7 @@ Harmony service for subsetting L2 data. l2ss-py supports:
 - Spatial subsetting
 - Temporal subsetting
 - Variable subsetting
+- Shapefile subsetting
```
Rather than adding a new item to this list I think we should do something like
> Spatial subsetting (via Bounding Box or Shapefile)

To make it more clear that it is still a spatial subset

frankinspace replied:
Just saw your other comments. Maybe we need to update documentation to specify GeoJSON is supported?

skorper commented:
@frankinspace I've updated the documentation to explain that both shapefiles and geojson files are supported.  I also added a `--shapefile` option to the l2ss-py CLI.
